From 3844ee49877b8300e2717f6a46e77b085f11c1ef Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 28 Jan 2021 17:40:29 +0200
Subject: [PATCH] [SW-28801] gaudi: reset: align halt engines

Align the halt engines to the latest good reset flow.
Move the gaudi_nics_handle() to the start of halt engines.

Change-Id: Idfe2cec7da1a18329fcee40b35ddc9a61b6cfed3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/reset.c b/gaudi/reset.c
index 334a38d..efbc0c6 100644
--- a/gaudi/reset.c
+++ b/gaudi/reset.c
@@ -342,6 +342,8 @@
 
 static void gaudi_halt_engines(void)
 {
+	gaudi_nics_handle();
+
 	gaudi_disable_clock_gating();
 
 	/* stop Rx/Tx here to drain any queued packets */
@@ -368,8 +370,6 @@
 	gaudi_disable_pci_dma_qmans();
 
 	gaudi_disable_timestamp();
-
-	gaudi_nics_handle();
 }
 
 static void gaudi_hw_fini(void)
